src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -231,32 +181,27 @@ internal T GetValue<T>(string key)
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <param name="key"></param>
-        /// <param name="throwEx"></param>
+        /// <param name="throwEx">if this is called from TryGetValue then we don't want to throw.</param>
         /// <param name="found"></param>
         /// <returns></returns>
         internal T GetValue<T>(string key, bool throwEx, out bool found)
         {
             found = _jsonClaims.TryGetValue(key, out object obj);
 
             if (!found)
-            {
-                if (throwEx)
-                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key)));
-                else
-                    return default;
-            }
+                return throwEx ? throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key))) : default;
 
             if (obj == null)
                 if (typeof(T) == typeof(object) || typeof(T).IsClass || Nullable.GetUnderlyingType(typeof(T)) != null)
-                {
                     return (T)(object)null;
-                }
                 else
-                {
-                    found = false;
                     return default;
-                }
 
+            // When the JsonClaimSet is created JsonArray and JsonObject are stored as JsonElement's
+            if (obj is JsonElement jsonElement)
+                return (T)JsonSerializerPrimitives.CreateTypeFromJsonElement<T>(jsonElement);
```

Comment:
CreateTypeFromJsonElement can throw an exception which does not respect throwEx, and it will bubble in a TryGet method. I think that would be unexpected behavior.

Reply:
Fixed.

Reply:
fixed. Renamed TryCreateType